src/routes/blog/index.json.ts

```diff
@@ -1,8 +1,14 @@
-import type { RequestHandlerOutput } from '@sveltejs/kit/types/internal';
-import { fetchNewsletters } from './_query';
+import type { RequestHandlerOutput, RequestEvent } from '@sveltejs/kit/types/internal';
+import { fetchNewsletters, NewsletterFetchOptions } from './_query';
 
-export async function get(): Promise<RequestHandlerOutput> {
-  return new Response(JSON.stringify(await fetchNewsletters()), {
+export async function get(event: RequestEvent): Promise<RequestHandlerOutput> {
+  const fetchOptions: NewsletterFetchOptions = { labels: [] };
+
+  if (event.url.searchParams.has('l') === true && event.url.searchParams.get('l').length > 0) {
+    fetchOptions.labels = event.url.searchParams.get('l').split(',');
+  }
```

Comment:
Nit: Simplification of conditional statement.
```suggestion
  const fetchOptions: NewsletterFetchOptions = { labels: [] };
  const rawLabels = event.url.searchParams.get('l');

  if (rawLabels?.length > 0) {
    fetchOptions.labels = rawLabels.split(',');
  }
```